src/dns/server.rs

```diff
@@ -1193,12 +1201,20 @@ mod tests {
 
         // Create and start the server.
         let domain = "cluster.local".to_string();
-        let addr = new_socket_addr().await;
         let state = state();
         let forwarder = Arc::new(SystemForwarder::new().unwrap());
-        let server = Server::new(domain, addr, NW1, state, forwarder, test_metrics())
-            .await
-            .unwrap();
+        let server = Server::new(
+            domain,
+            SocketAddr::from_str("127.0.0.1:0")
```

Comment:
nit: I think you can do something like `SocketAddr::from(([127,0,0,1], 0))` or similar but I forget the syntax

Reply:
Great point. It's marginally less readable but because it's from simpler types it can be compile-time checked. I like that more but I missed the comment before it auto-merged. I'll do a quick follow up.